instrumentation/CONTRIBUTING.md

```diff
@@ -0,0 +1,483 @@
+
+1. Generate an instrumentation gem skeleton
+2. Implement the instrumentation library including comprehensive automated tests
+3. Add the instrumentation library to the appropriate CI workflow
+4. Include documentation for your instrumentation
+    * Document all instrumentation-specific configuration options in the `README.md` and `yardoc` comments
+    * Document all semantic conventions used by the instrumentation in the `README.md`
+    * Provide executable examples in an `examples` directory
+5. Submit a pull request
+
+## Generate the Gem
+
+This repository contains a script to generate a new instrumentation library.
+
+The snippet below demonstrates how to generate a an instrumentation for the `werewolf` gem, starting from the repository root.
+
+```console
+
+$bash opentelemetry-ruby-contrib> bin/instrumentation_generator werewolf
+
+```
+
+The output of the generator shows that it creates a new directory in the `instrumentation` directory using the name of the gem:
+
+``` console
+
+🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
+
+ WARNING: Your gem will *NOT* be tested until you add it to the CI workflows in `.github/workflows/ci.yml`!!
+
+🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
+
+      create  instrumentation/werewolf/.rubocop.yml
+      create  instrumentation/werewolf/.yardopts
+      create  instrumentation/werewolf/Appraisals
+      create  instrumentation/werewolf/CHANGELOG.md
+      create  instrumentation/werewolf/Gemfile
+      create  instrumentation/werewolf/LICENSE
+      create  instrumentation/werewolf/opentelemetry-instrumentation-werewolf.gemspec
+      create  instrumentation/werewolf/Rakefile
+      create  instrumentation/werewolf/README.md
+      create  instrumentation/werewolf/lib/opentelemetry-instrumentation-werewolf.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf/instrumentation.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf/version.rb
+      create  instrumentation/werewolf/test/test_helper.rb
+      create  instrumentation/werewolf/test/opentelemetry/instrumentation/werewolf/instrumentation_test.rb
+      insert  .toys/.data/releases.yml
+      insert  instrumentation/all/Gemfile
+      insert  instrumentation/all/opentelemetry-instrumentation-all.gemspec
+      insert  instrumentation/all/lib/opentelemetry/instrumentation/all.rb
+
+```
+
+## Implementation guidelines
+
+The original design and implementation of this project was heavily influenced by Datadog's `dd-trace-rb` project. You may refer to the [Datadog Porting Guide](datadog-porting-guide.md) as a reference for implementing instrumentations, however the following guidelines are specific to OpenTelemetry Ruby:
+
+* Use `OpenTelemetry::Instrumentation::Base`
```

Comment:
Since this group of bullets matches the headings for the sections below, what do you think about adding anchor links?

Reply:
GitHub MD auto generates anchor links to a heading when you click it. Do you mean having sections use links to refer to each other?

Reply:
That's right, I forgot the links were automatically generated! So it would be linking whatever the auto-generated value is for the heading to the bullet with the same heading: 

```markdown
* [Use `OpenTelemetry::Instrumentation::Base`](#anchor-link-to-Use-OpenTelemetry-Instrumentation-Base-heading)
```

Reply:
I see. Treating that bulleted list more like a sub table of contents.